File: frontend/src/lib/api/qaPlan.ts
// QA 計畫管理 API（稽查報告、不符合事項、SOP 文件、稽查排程）

import api from './client'

// ========== Types ==========

export type QaInspectionType = 'protocol' | 'equipment' | 'facility' | 'training' | 'general'
export type QaInspectionStatus = 'draft' | 'submitted' | 'closed'
export type QaItemResult = 'pass' | 'fail' | 'not_applicable'
export type NcSeverity = 'critical' | 'major' | 'minor'
export type NcSource = 'inspection' | 'observation' | 'external_audit' | 'self_report'
export type NcStatus = 'open' | 'in_progress' | 'pending_verification' | 'closed'
export type CapaActionType = 'corrective' | 'preventive'
export type CapaStatus = 'open' | 'in_progress' | 'completed' | 'verified'
export type SopStatus = 'draft' | 'active' | 'obsolete'
export type QaScheduleType = 'annual' | 'periodic' | 'ad_hoc'
export type QaScheduleStatus = 'planned' | 'in_progress' | 'completed' | 'cancelled'
export type QaScheduleItemStatus = 'planned' | 'in_progress' | 'completed' | 'cancelled' | 'overdue'

export interface QaInspectionWithInspector {
  id: string
  inspection_number: string
  title: string
  inspection_type: QaInspectionType
  inspection_date: string
  inspector_id: string
  inspector_name: string
  related_entity_type: string | null
  related_entity_id: string | null
  status: QaInspectionStatus
  findings: string | null
  conclusion: string | null
  created_at: string
  updated_at: string
}

export interface QaInspectionItem {
  id: string
  inspection_id: string
  item_order: number
  description: string
  result: QaItemResult
  remarks: string | null
  created_at: string
}

export interface QaInspectionDetail extends QaInspectionWithInspector {
  items: QaInspectionItem[]
}

export interface QaNonConformanceWithDetails {
  id: string
  nc_number: string
  title: string
  description: string
  severity: NcSeverity
  source: NcSource
  related_inspection_id: string | null
  assignee_id: string | null
  assignee_name: string | null
  due_date: string | null
  status: NcStatus
  root_cause: string | null
  closure_notes: string | null
  closed_at: string | null
  created_by: string
  creator_name: string
  created_at: string
  updated_at: string
}

export interface QaCapa {
  id: string
  nc_id: string
  action_type: CapaActionType
  description: string
  assignee_id: string | null
  due_date: string | null
  completed_at: string | null
  status: CapaStatus
  created_at: string
  updated_at: string
}

export interface NcDetail extends QaNonConformanceWithDetails {
  capa: QaCapa[]
}

export interface QaSopDocumentWithAck {
  id: string
  document_number: string
  title: string
  version: string
  category: string | null
  file_path: string | null
  effective_date: string | null
  review_date: string | null
  status: SopStatus
  description: string | null
  created_by: string
  creator_name: string
  acknowledged_by_me: boolean
  ack_count: number
  created_at: string
  updated_at: string
}

export interface QaAuditSchedule {
  id: string
  year: number
  title: string
  schedule_type: QaScheduleType
  description: string | null
  status: QaScheduleStatus
  created_by: string
  created_at: string
  updated_at: string
}

export interface QaScheduleItem {
  id: string
  schedule_id: string
  inspection_type: QaInspectionType
  title: string
  planned_date: string
  actual_date: string | null
  responsible_person_id: string | null
  responsible_name: string | null
  related_inspection_id: string | null
  status: QaScheduleItemStatus
  notes: string | null
  created_at: string
  updated_at: string
}

export interface QaScheduleDetail extends QaAuditSchedule {
  items: QaScheduleItem[]
}

// ========== Inspection API ==========

export const listInspections = async (params?: {
  page?: number
  page_size?: number
  inspection_type?: string
  status?: string
}) => {
  const res = await api.get<{ data: QaInspectionWithInspector[] }>('/qau/inspections', { params })
  return res.data.data
}

export const getInspection = async (id: string) => {
  const res = await api.get<QaInspectionDetail>(`/qau/inspections/${id}`)
  return res.data
}

export const createInspection = async (payload: {
  title: string
  inspection_type: QaInspectionType
  inspection_date: string
  related_entity_type?: string
  related_entity_id?: string
  findings?: string
  conclusion?: string
  items: { item_order: number; description: string; result: QaItemResult; remarks?: string }[]
}) => {
  const res = await api.post<QaInspectionDetail>('/qau/inspections', payload)
  return res.data
}

export const updateInspection = async (
  id: string,
  payload: Partial<{
    title: string
    inspection_date: string
    findings: string
    conclusion: string
    status: QaInspectionStatus
    items: { item_order: number; description: string; result: QaItemResult; remarks?: string }[]
  }>
) => {
  const res = await api.put<QaInspectionDetail>(`/qau/inspections/${id}`, payload)
  return res.data
}

// ========== Non-Conformance API ==========

export const listNonConformances = async (params?: {
  page?: number
  page_size?: number
  severity?: string
  status?: string
}) => {
  const res = await api.get<{ data: QaNonConformanceWithDetails[] }>('/qau/non-conformances', { params })
  return res.data.data
}

export const getNonConformance = async (id: string) => {
  const res = await api.get<NcDetail>(`/qau/non-conformances/${id}`)
  return res.data
}

export const createNonConformance = async (payload: {
  title: string
  description: string
  severity: NcSeverity
  source: NcSource
  related_inspection_id?: string
  assignee_id?: string
  due_date?: string
}) => {
  const res = await api.post<NcDetail>('/qau/non-conformances', payload)
  return res.data
}

export const updateNonConformance = async (
  id: string,
  payload: Partial<{
    title: string
    description: string
    assignee_id: string
    due_date: string
    status: NcStatus
    root_cause: string
    closure_notes: string
  }>
) => {
  const res = await api.put<NcDetail>(`/qau/non-conformances/${id}`, payload)
  return res.data
}

export const createCapa = async (
  ncId: string,
  payload: { action_type: CapaActionType; description: string; assignee_id?: string; due_date?: string }
) => {
  const res = await api.post<QaCapa>(`/qau/non-conformances/${ncId}/capa`, payload)
  return res.data
}

export const updateCapa = async (
  ncId: string,
  capaId: string,
  payload: Partial<{ description: string; assignee_id: string; due_date: string; status: CapaStatus }>
) => {
  const res = await api.put<QaCapa>(`/qau/non-conformances/${ncId}/capa/${capaId}`, payload)
  return res.data
}

// ========== SOP API ==========

export const listSopDocuments = async (params?: {
  page?: number
  page_size?: number
  status?: string
  category?: string
}) => {
  const res = await api.get<{ data: QaSopDocumentWithAck[] }>('/qau/sop', { params })
  return res.data.data
}

export const getSopDocument = async (id: string) => {
  const res = await api.get<QaSopDocumentWithAck>(`/qau/sop/${id}`)
  return res.data
}

export const createSopDocument = async (payload: {
  title: string
  version: string
  category?: string
  file_path?: string
  effective_date?: string
  review_date?: string
  description?: string
}) => {
  const res = await api.post<QaSopDocumentWithAck>('/qau/sop', payload)
  return res.data
}

export const updateSopDocument = async (
  id: string,
  payload: Partial<{
    title: string
    version: string
    category: string
    file_path: string
    effective_date: string
    review_date: string
    status: SopStatus
    description: string
  }>
) => {
  const res = await api.put<QaSopDocumentWithAck>(`/qau/sop/${id}`, payload)
  return res.data
}

export const acknowledgeSop = async (id: string) => {
  const res = await api.post<QaSopDocumentWithAck>(`/qau/sop/${id}/acknowledge`)
  return res.data
}

// ========== Schedule API ==========

export const listSchedules = async (params?: { year?: number; status?: string }) => {
  const res = await api.get<{ data: QaAuditSchedule[] }>('/qau/schedules', { params })
  return res.data.data
}

export const getSchedule = async (id: string) => {
  const res = await api.get<QaScheduleDetail>(`/qau/schedules/${id}`)
  return res.data
}

export const createSchedule = async (payload: {
  year: number
  title: string
  schedule_type: QaScheduleType
  description?: string
  items: {
    inspection_type: QaInspectionType
    title: string
    planned_date: string
    responsible_person_id?: string
    notes?: string
  }[]
}) => {
  const res = await api.post<QaScheduleDetail>('/qau/schedules', payload)
  return res.data
}

export const updateSchedule = async (
  id: string,
  payload: Partial<{ title: string; description: string; status: QaScheduleStatus }>
) => {
  const res = await api.put<QaScheduleDetail>(`/qau/schedules/${id}`, payload)
  return res.data
}

export const updateScheduleItem = async (
  scheduleId: string,
  itemId: string,
  payload: Partial<{
    actual_date: string
    responsible_person_id: string
    related_inspection_id: string
    status: QaScheduleItemStatus
    notes: string
  }>
) => {
  const res = await api.put<QaScheduleDetail>(`/qau/schedules/${scheduleId}/items/${itemId}`, payload)
  return res.data
}

File: backend/src/models/qa_plan.rs
// QA 計畫管理 Models（稽查報告、不符合事項、SOP 文件、稽查排程）

use chrono::{DateTime, NaiveDate, Utc};
use serde::{Deserialize, Serialize};
use sqlx::{FromRow, Type};
use uuid::Uuid;
use validator::Validate;

// ========== Enums ==========

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "qa_inspection_type", rename_all = "snake_case")]
pub enum QaInspectionType {
    #[serde(rename = "protocol")]
    Protocol,
    #[serde(rename = "equipment")]
    Equipment,
    #[serde(rename = "facility")]
    Facility,
    #[serde(rename = "training")]
    Training,
    #[serde(rename = "general")]
    General,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "qa_inspection_status", rename_all = "snake_case")]
pub enum QaInspectionStatus {
    #[serde(rename = "draft")]
    Draft,
    #[serde(rename = "submitted")]
    Submitted,
    #[serde(rename = "closed")]
    Closed,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "qa_item_result", rename_all = "snake_case")]
pub enum QaItemResult {
    #[serde(rename = "pass")]
    Pass,
    #[serde(rename = "fail")]
    Fail,
    #[serde(rename = "not_applicable")]
    NotApplicable,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "nc_severity", rename_all = "snake_case")]
pub enum NcSeverity {
    #[serde(rename = "critical")]
    Critical,
    #[serde(rename = "major")]
    Major,
    #[serde(rename = "minor")]
    Minor,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "nc_source", rename_all = "snake_case")]
pub enum NcSource {
    #[serde(rename = "inspection")]
    Inspection,
    #[serde(rename = "observation")]
    Observation,
    #[serde(rename = "external_audit")]
    ExternalAudit,
    #[serde(rename = "self_report")]
    SelfReport,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "nc_status", rename_all = "snake_case")]
pub enum NcStatus {
    #[serde(rename = "open")]
    Open,
    #[serde(rename = "in_progress")]
    InProgress,
    #[serde(rename = "pending_verification")]
    PendingVerification,
    #[serde(rename = "closed")]
    Closed,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "capa_action_type", rename_all = "snake_case")]
pub enum CapaActionType {
    #[serde(rename = "corrective")]
    Corrective,
    #[serde(rename = "preventive")]
    Preventive,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "capa_status", rename_all = "snake_case")]
pub enum CapaStatus {
    #[serde(rename = "open")]
    Open,
    #[serde(rename = "in_progress")]
    InProgress,
    #[serde(rename = "completed")]
    Completed,
    #[serde(rename = "verified")]
    Verified,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "sop_status", rename_all = "snake_case")]
pub enum SopStatus {
    #[serde(rename = "draft")]
    Draft,
    #[serde(rename = "active")]
    Active,
    #[serde(rename = "obsolete")]
    Obsolete,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "qa_schedule_type", rename_all = "snake_case")]
pub enum QaScheduleType {
    #[serde(rename = "annual")]
    Annual,
    #[serde(rename = "periodic")]
    Periodic,
    #[serde(rename = "ad_hoc")]
    AdHoc,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "qa_schedule_status", rename_all = "snake_case")]
pub enum QaScheduleStatus {
    #[serde(rename = "planned")]
    Planned,
    #[serde(rename = "in_progress")]
    InProgress,
    #[serde(rename = "completed")]
    Completed,
    #[serde(rename = "cancelled")]
    Cancelled,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Type)]
#[sqlx(type_name = "qa_schedule_item_status", rename_all = "snake_case")]
pub enum QaScheduleItemStatus {
    #[serde(rename = "planned")]
    Planned,
    #[serde(rename = "in_progress")]
    InProgress,
    #[serde(rename = "completed")]
    Completed,
    #[serde(rename = "cancelled")]
    Cancelled,
    #[serde(rename = "overdue")]
    Overdue,
}

// ========== 稽查報告 ==========

#[derive(Debug, Serialize, FromRow)]
pub struct QaInspection {
    pub id: Uuid,
    pub inspection_number: String,
    pub title: String,
    pub inspection_type: QaInspectionType,
    pub inspection_date: NaiveDate,
    pub inspector_id: Uuid,
    pub related_entity_type: Option<String>,
    pub related_entity_id: Option<Uuid>,
    pub status: QaInspectionStatus,
    pub findings: Option<String>,
    pub conclusion: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, FromRow)]
pub struct QaInspectionWithInspector {
    pub id: Uuid,
    pub inspection_number: String,
    pub title: String,
    pub inspection_type: QaInspectionType,
    pub inspection_date: NaiveDate,
    pub inspector_id: Uuid,
    pub inspector_name: String,
    pub related_entity_type: Option<String>,
    pub related_entity_id: Option<Uuid>,
    pub status: QaInspectionStatus,
    pub findings: Option<String>,
    pub conclusion: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, FromRow)]
pub struct QaInspectionItem {
    pub id: Uuid,
    pub inspection_id: Uuid,
    pub item_order: i32,
    pub description: String,
    pub result: QaItemResult,
    pub remarks: Option<String>,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Serialize)]
pub struct QaInspectionDetail {
    #[serde(flatten)]
    pub inspection: QaInspectionWithInspector,
    pub items: Vec<QaInspectionItem>,
}

#[derive(Debug, Deserialize, Validate)]
pub struct CreateInspectionRequest {
    #[validate(length(min = 1, max = 255))]
    pub title: String,
    pub inspection_type: QaInspectionType,
    pub inspection_date: NaiveDate,
    pub related_entity_type: Option<String>,
    pub related_entity_id: Option<Uuid>,
    pub findings: Option<String>,
    pub conclusion: Option<String>,
    pub items: Vec<CreateInspectionItemRequest>,
}

#[derive(Debug, Deserialize)]
pub struct CreateInspectionItemRequest {
    pub item_order: i32,
    pub description: String,
    pub result: QaItemResult,
    pub remarks: Option<String>,
}

#[derive(Debug, Deserialize, Validate)]
pub struct UpdateInspectionRequest {
    #[validate(length(min = 1, max = 255))]
    pub title: Option<String>,
    pub inspection_date: Option<NaiveDate>,
    pub findings: Option<String>,
    pub conclusion: Option<String>,
    pub status: Option<QaInspectionStatus>,
    pub items: Option<Vec<CreateInspectionItemRequest>>,
}

#[derive(Debug, Deserialize, Default)]
pub struct InspectionQuery {
    pub page: Option<i64>,
    pub page_size: Option<i64>,
    pub inspection_type: Option<String>,
    pub status: Option<String>,
}

// ========== 不符合事項 ==========

#[derive(Debug, Serialize, FromRow)]
pub struct QaNonConformance {
    pub id: Uuid,
    pub nc_number: String,
    pub title: String,
    pub description: String,
    pub severity: NcSeverity,
    pub source: NcSource,
    pub related_inspection_id: Option<Uuid>,
    pub assignee_id: Option<Uuid>,
    pub due_date: Option<NaiveDate>,
    pub status: NcStatus,
    pub root_cause: Option<String>,
    pub closure_notes: Option<String>,
    pub closed_at: Option<DateTime<Utc>>,
    pub created_by: Uuid,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, FromRow)]
pub struct QaNonConformanceWithDetails {
    pub id: Uuid,
    pub nc_number: String,
    pub title: String,
    pub description: String,
    pub severity: NcSeverity,
    pub source: NcSource,
    pub related_inspection_id: Option<Uuid>,
    pub assignee_id: Option<Uuid>,
    pub assignee_name: Option<String>,
    pub due_date: Option<NaiveDate>,
    pub status: NcStatus,
    pub root_cause: Option<String>,
    pub closure_notes: Option<String>,
    pub closed_at: Option<DateTime<Utc>>,
    pub created_by: Uuid,
    pub creator_name: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, FromRow)]
pub struct QaCapa {
    pub id: Uuid,
    pub nc_id: Uuid,
    pub action_type: CapaActionType,
    pub description: String,
    pub assignee_id: Option<Uuid>,
    pub due_date: Option<NaiveDate>,
    pub completed_at: Option<DateTime<Utc>>,
    pub status: CapaStatus,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize)]
pub struct NcDetail {
    #[serde(flatten)]
    pub nc: QaNonConformanceWithDetails,
    pub capa: Vec<QaCapa>,
}

#[derive(Debug, Deserialize, Validate)]
pub struct CreateNcRequest {
    #[validate(length(min = 1, max = 255))]
    pub title: String,
    pub description: String,
    pub severity: NcSeverity,
    pub source: NcSource,
    pub related_inspection_id: Option<Uuid>,
    pub assignee_id: Option<Uuid>,
    pub due_date: Option<NaiveDate>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateNcRequest {
    pub title: Option<String>,
    pub description: Option<String>,
    pub assignee_id: Option<Uuid>,
    pub due_date: Option<NaiveDate>,
    pub status: Option<NcStatus>,
    pub root_cause: Option<String>,
    pub closure_notes: Option<String>,
}

#[derive(Debug, Deserialize, Validate)]
pub struct CreateCapaRequest {
    pub action_type: CapaActionType,
    #[validate(length(min = 1))]
    pub description: String,
    pub assignee_id: Option<Uuid>,
    pub due_date: Option<NaiveDate>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateCapaRequest {
    pub description: Option<String>,
    pub assignee_id: Option<Uuid>,
    pub due_date: Option<NaiveDate>,
    pub status: Option<CapaStatus>,
}

#[derive(Debug, Deserialize, Default)]
pub struct NcQuery {
    pub page: Option<i64>,
    pub page_size: Option<i64>,
    pub severity: Option<String>,
    pub status: Option<String>,
}

// ========== SOP 文件 ==========

#[derive(Debug, Serialize, FromRow)]
pub struct QaSopDocument {
    pub id: Uuid,
    pub document_number: String,
    pub title: String,
    pub version: String,
    pub category: Option<String>,
    pub file_path: Option<String>,
    pub effective_date: Option<NaiveDate>,
    pub review_date: Option<NaiveDate>,
    pub status: SopStatus,
    pub description: Option<String>,
    pub created_by: Uuid,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, FromRow)]
pub struct QaSopDocumentWithAck {
    pub id: Uuid,
    pub document_number: String,
    pub title: String,
    pub version: String,
    pub category: Option<String>,
    pub file_path: Option<String>,
    pub effective_date: Option<NaiveDate>,
    pub review_date: Option<NaiveDate>,
    pub status: SopStatus,
    pub description: Option<String>,
    pub created_by: Uuid,
    pub creator_name: String,
    pub acknowledged_by_me: bool,
    pub ack_count: i64,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Deserialize, Validate)]
pub struct CreateSopRequest {
    #[validate(length(min = 1, max = 255))]
    pub title: String,
    #[validate(length(min = 1, max = 20))]
    pub version: String,
    pub category: Option<String>,
    pub file_path: Option<String>,
    pub effective_date: Option<NaiveDate>,
    pub review_date: Option<NaiveDate>,
    pub description: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateSopRequest {
    pub title: Option<String>,
    pub version: Option<String>,
    pub category: Option<String>,
    pub file_path: Option<String>,
    pub effective_date: Option<NaiveDate>,
    pub review_date: Option<NaiveDate>,
    pub status: Option<SopStatus>,
    pub description: Option<String>,
}

#[derive(Debug, Deserialize, Default)]
pub struct SopQuery {
    pub page: Option<i64>,
    pub page_size: Option<i64>,
    pub status: Option<String>,
    pub category: Option<String>,
}

// ========== 稽查排程 ==========

#[derive(Debug, Serialize, FromRow)]
pub struct QaAuditSchedule {
    pub id: Uuid,
    pub year: i32,
    pub title: String,
    pub schedule_type: QaScheduleType,
    pub description: Option<String>,
    pub status: QaScheduleStatus,
    pub created_by: Uuid,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, FromRow)]
pub struct QaScheduleItem {
    pub id: Uuid,
    pub schedule_id: Uuid,
    pub inspection_type: QaInspectionType,
    pub title: String,
    pub planned_date: NaiveDate,
    pub actual_date: Option<NaiveDate>,
    pub responsible_person_id: Option<Uuid>,
    pub responsible_name: Option<String>,
    pub related_inspection_id: Option<Uuid>,
    pub status: QaScheduleItemStatus,
    pub notes: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize)]
pub struct QaScheduleDetail {
    #[serde(flatten)]
    pub schedule: QaAuditSchedule,
    pub items: Vec<QaScheduleItem>,
}

#[derive(Debug, Deserialize, Validate)]
pub struct CreateScheduleRequest {
    pub year: i32,
    #[validate(length(min = 1, max = 255))]
    pub title: String,
    pub schedule_type: QaScheduleType,
    pub description: Option<String>,
    pub items: Vec<CreateScheduleItemRequest>,
}

#[derive(Debug, Deserialize)]
pub struct CreateScheduleItemRequest {
    pub inspection_type: QaInspectionType,
    pub title: String,
    pub planned_date: NaiveDate,
    pub responsible_person_id: Option<Uuid>,
    pub notes: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateScheduleItemRequest {
    pub actual_date: Option<NaiveDate>,
    pub responsible_person_id: Option<Uuid>,
    pub related_inspection_id: Option<Uuid>,
    pub status: Option<QaScheduleItemStatus>,
    pub notes: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateScheduleRequest {
    pub title: Option<String>,
    pub description: Option<String>,
    pub status: Option<QaScheduleStatus>,
}

#[derive(Debug, Deserialize, Default)]
pub struct ScheduleQuery {
    pub year: Option<i32>,
    pub status: Option<String>,
}
